backend/src/main/java/pl/gddkia/estimate/EstimateServiceImpl.java

```diff
@@ -143,29 +143,34 @@ public List<EstimateRest> getEstimate() {
         return estimateRepository.findAll()
                 .stream()
                 .map(estimate -> new EstimateRest(
-                        estimate.getId(),
-                        estimate.getContractName(),
-                        estimate.getDateFrom().toString(),
-                        estimate.getDateTo().toString(),
-                        estimate.getRegion().getRegionName(),
-                        estimate.getRegion().getBranch().getBranchName(),
-                        groupRepository.findAllByEstimateId(estimate.getId())
-                                .stream()
-                                .collect(
-                                        Collectors.toMap(
-                                                Group::getGroupName,
-                                                group -> group.getJobs().stream().map(jobs -> new JobRest(
-                                                        jobs.getSST(),
-                                                        jobs.getDescription(),
-                                                        jobs.getUnit(),
-                                                        jobs.getCostEstimate(),
-                                                        jobs.getQuantity(),
-                                                        jobs.getSubType()
-                                                )).toList()
+                                estimate.getId(),
+                                estimate.getContractName(),
+                                estimate.getDateFrom().toString(),
+                                estimate.getDateTo().toString(),
+                                estimate.getRegion().getRegionName(),
+                                estimate.getRegion().getBranch().getBranchName(),
+                                groupRepository.findAllByEstimateId(estimate.getId())
```

Comment:
We can split code here into smaller parts to make it overall easier to debug and prettier .

Code examples: `public List<EstimateRest> getAllEstimates() {
    return estimateRepository.findAll()
            .stream()
            .map(this::mapToEstimateRest)
            .collect(Collectors.toList());
}

private EstimateRest mapToEstimateRest(Estimate estimate) {
    Map<String, List<JobRest>> groupedJobs = groupRepository.findAllByEstimateId(estimate.getId())
            .stream()
            .collect(Collectors.groupingBy(Group::getGroupName,
                    Collectors.mapping(this::mapToJobRest, Collectors.toList())));

    return new EstimateRest(
            estimate.getId(),
            estimate.getContractName(),
            estimate.getDateFrom().toString(),
            estimate.getDateTo().toString(),
            estimate.getRegion().getRegionName(),
            estimate.getRegion().getBranch().getBranchName(),
            groupedJobs
    );
}

private JobRest mapToJobRest(Group group) {
    return group.getJobs()
            .stream()
            .map(job -> new JobRest(
                    job.getSST(),
                    job.getDescription(),
                    job.getUnit(),
                    job.getCostEstimate(),
                    job.getQuantity(),
                    job.getSubType()
            ))
            .toList();
}
`

Reply:
There's no need to do this in this PR, we can change more code like this after ofc.

Reply:
For now, i will leave those crazy looking mappings for feature refactoring